src/gui/qgsmaptoolcapture.cpp

```diff
@@ -1391,6 +1391,27 @@ void QgsMapToolCapture::cadCanvasReleaseEvent( QgsMapMouseEvent *e )
       }
       else
       {
+
+        //does compoundcurve contain circular strings?
+        //does provider support circular strings?
+        if ( QgsVectorLayer *vlayer = qobject_cast<QgsVectorLayer *>( layer() ) )
+        {
+          const bool hasCurvedSegments = captureCurve()->hasCurvedSegments();
+          const bool providerSupportsCurvedSegments = vlayer->dataProvider()->capabilities() & QgsVectorDataProvider::CircularGeometries;
+
+          if ( hasCurvedSegments && providerSupportsCurvedSegments )
+          {
+            curveToAdd = captureCurve()->clone();
```

Comment:
This introduces a leak, because the existing value of curveToAdd is not cleaned up.

It would be better to make curveToAdd a unique ptr to avoid this

Reply:
@lbartoletti, shouldn't it be fixed in master, since such code was introduced with PR https://github.com/qgis/QGIS/pull/55371?

Reply:
Yes, I'll try to fix at the beginning of the next week.